metis-indexing/src/main/java/eu/europeana/indexing/mongo/property/MongoPropertyUpdaterImpl.java

```diff
@@ -266,12 +266,14 @@ private static boolean equals(AbstractEdmEntity entity1, AbstractEdmEntity entit
   @Override
   public void updateWebResourceMetaInfo(Function<T, WebResourceMetaInfo> getter,
       Function<T, WebResourceInformation> ancestorInfoGetter,
-      Supplier<MongoObjectUpdater<WebResourceMetaInfoImpl, WebResourceInformation>> updaterSupplier) {
+      Supplier<MongoObjectManager<WebResourceMetaInfoImpl, WebResourceInformation>> updaterSupplier) {
     final WebResourceMetaInfo entity = Optional.of(updated).map(getter).orElse(null);
     final WebResourceInformation ancestorInformation = ancestorInfoGetter.apply(updated);
-    if (entity != null) {
-      updaterSupplier.get()
-          .update((WebResourceMetaInfoImpl) entity, ancestorInformation, null, null, mongoServer);
+    if (entity == null) {
+      updaterSupplier.get().delete(ancestorInformation, mongoServer);
```

Comment:
Should we be checking nullity on ancestorInformation too?
I know that what is provided is not null, but for extra safety?

Reply:
I could, but it would be redundant, I think? Currently, if it is null it would result in a NullPointerException in the code that is called (either deletion or update). And that's what we'd expect, I think. So all we would do is create some lines of code to check and create an exception that you would get anyway.
Or am I missing something?

Reply:
No you  are not, I just thought for extra security but I guess you're right and we are safe here.